### PR TITLE
Add support for docker and Nethermind client to the simulation tests 

### LINKED ISSUE #4275: Sim tests overhaul
**Background**

_Sim test = running a full devnet for some time_

Sim tests are crucial to test blockchain nodes due to their complexity. Unit tests or e2e test never capture the complexity of the system when it's running at full capacity. Even then running a full devnet in CI is not sufficient to capture all bugs since some only appear under certain network size and / or conditions.

The scope and goals of sim tests in this repo should be:
- [x] Run small devnet for a relatively short period of time.
- [x] Capture a large amount of metrics to ensure good performance on all fronts.
- [x] Do so every commit, or at least on every commit of the main branch

Sim tests are (and will be) complemented by:
- [x] ~~Kurtosis testing: Run small devnets for short time but with all other client combinations. Medium level of assertions~~ Moved to #4772
- [x] ~~Hive testing: Run small devnets for short time but with unusual network conditions. Low level of assertions https://github.com/ChainSafe/lodestar/issues/4240~~ Moved to #4773
- [x] Deployment on Prater (beta testing): Run big devnet for long periods of time, but only infrequently before releases

**TODO**

Current sim test implementation is very old an subpar. It could be structured in a way that more modular, mantainable, and stable. However the first important step is to add assertions to cover the missing tick in the scope outline above:

- [x] Add proper assertions to sim tests https://github.com/ChainSafe/lodestar/issues/4276
- [x] Migrate leftover sim tests from e2e, to make e2e tests more stable and fast
- [x] Consolidate sim tests into less runs: do lightclient, finality, sync tests in one run https://github.com/ChainSafe/lodestar/issues/3931
- [x] Enable multi-node multi-process testing
  - Attempt at implementing https://github.com/ChainSafe/lodestar/pull/2575
  - Blocking issue https://github.com/ChainSafe/lodestar/issues/2542
- [x] ~~If expensive at this point, run in self-deployed GA runner.~~ **EDIT (nov 11th)**: keep the run light to work in Github runners

**Progressive improvements**
- [x] Log std to file, disable log to file. Enable DEBUG=* for Lodestar
- [x] Run Geth and Nethermind and test going through the merge, until finalizing merge block + 2 epochs
- [x] Ensure all nodes are connected to each other https://github.com/ChainSafe/lodestar/issues/4645
- [x] Once nodes reach untilEpoch, keep the chain running and spawn extra nodes to test syncing
  - [x] Start 1 node from genesis and range sync to head
  - [x] Start 1 node with checkpoint sync and sync to head  
- [x] Update the test timeout to be calculated [deterministically](https://github.com/ChainSafe/lodestar/pull/4673#discussion_r1007250212)
- [x] Disable the file logging for lodestar by explicitly setting it to `none`
- [x] Update the sync tests approach by just testing the sync behavior and not waiting for a particular epoch.

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Add support for the docker and nethermind client to simulation tests. 

**Description**

- [x] Run Nethermind and test going through the merge, until finalizing merge block + 2 epochs

Partially Resolves: #4275 

**Steps to test or reproduce**

Run sim tests.

### CODE REVIEW

**Comment by nazarhussain** [comment]
While investing the errors on the PR.

```
Eph 4/4 0.122[node-1-cl-lodestar/chain] [31merror[39m: Block error slot=36 code=BLOCK_ERROR_EXECUTION_ERROR, execStatus=UNSAFE_OPTIMISTIC_STATUS, errorMessage=not safe to import SYNCING payload within 128 of currentSlot
Error: BLOCK_ERROR_EXECUTION_ERROR
    at verifyBlockExecutionPayload (file:///home/runner/work/lodestar/lodestar/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:307:27)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at verifyBlocksExecutionPayload (file:///home/runner/work/lodestar/lodestar/packages/beacon-node/src/chain/blocks/verifyBlocksExecutionPayloads.ts:151:28)
    at async Promise.all (index 2)
    at BeaconChain.verifyBlocksInEpoch (file:///home/runner/work/lodestar/lodestar/packages/beacon-node/src/chain/blocks/verifyBlock.ts:67:72)
    at BeaconChain.processBlocks (file:///home/runner/work/lodestar/lodestar/packages/beacon-node/src/chain/blocks/index.ts:71:68)
    at Timeout.JobItemQueue.runJob [as _onTimeout] (file:///home/runner/work/lodestar/lodestar/packages/beacon-node/src/util/queue/itemQueue.ts:92:22)
```

Found that: 

1. We had multiple Geth nodes running with mining mode. Both were proposing the conflicting blocks and caused this issue.

File: packages/cli/test/simulation/multi_fork.test.ts [hunk 1]
```diff
@@ -30,13 +42,14 @@ const env = SimulationEnvironment.initWithDefaults(
...
       BELLATRIX_FORK_EPOCH: bellatrixForkEpoch,
       GENESIS_DELAY: genesisSlotsDelay,
+      TERMINAL_TOTAL_DIFFICULTY: ttd,
     },
   },
   [
-    {id: "node-1", cl: CLClient.Lodestar, el: ELClient.Geth, keysCount: 32},
-    {id: "node-2", cl: CLClient.Lodestar, el: ELClient.Geth, keysCount: 32},
+    {id: "node-1", cl: CLClient.Lodestar, el: ELClient.Geth, keysCount: 32, remote: true, mining: true},
+    {id: "node-2", cl: CLClient.Lodestar, el: ELClient.Nethermind, keysCount: 32, remote: true},
```
Comment (dapplion) [inline packages/cli/test/simulation/multi_fork.test.ts:50]: Just have a 1 node with remote signer, pick the second to not be also the miner

Comment (nazarhussain) [reply packages/cli/test/simulation/multi_fork.test.ts:50]: I made the second node as remote signer. All others are not. And first node is the only miner in the set.

**Review by dapplion — APPROVED** [review]
Excellent!

**Review by wemeetagain — APPROVED** [review]

**Comment by nazarhussain** [comment]
For future reference the last 10 runs for sim tests after this PR passes successfully. 

image

**Pull request merged**